packages/svelte/src/runtime/internal/Component.js

```diff
@@ -21,10 +21,13 @@ import { transition_in } from './transitions.js';
 
 /** @returns {void} */
 export function bind(component, name, callback) {
-	const index = component.$$.props[name];
-	if (index !== undefined) {
-		component.$$.bound[index] = callback;
-		callback(component.$$.ctx[index]);
+	const i = component.$$.props[name];
+	if (i !== undefined) {
+		let dirty = false;
+		if (component.$$.bound[i]) dirty = true;
+		component.$$.bound[i] = callback;
+		// first binding call, if child value is not yet dirty, skip to prevent unnecessary backflow
+		callback(component.$$.ctx[i], /** skip_binding */ !dirty);
```

Comment:
The `bind` runtime function triggers the first backflow during `flush`, here I pass a `skip_binding` flag to `child_value_binding`. The flag is determined by `$$.bound[i]` dirty mark. This takes care of case 2 and 3.

Case 2: Child does not touch passed-in prop value
Case 3: Child modifies the value inside `$:` reactive statements